src/cljs/rems/administration/create_form.cljs

```diff
@@ -177,6 +177,47 @@
                    :error-handler status-modal/common-error-handler!}))
      {:db (assoc db ::form-errors form-errors)})))
 
+;;;; preview auto-scrolling
+
+(defn visibility-ratio [element]
+  (let [bounds (.getBoundingClientRect element)]
+    (cond (<= (.-bottom bounds) 0)
+          0
+          (>= (.-top bounds) 0)
+          1
+          :else
+          (/ (.-bottom bounds) (.-height bounds)))))
+
+(defn true-height [element]
+  (let [style (.getComputedStyle js/window element)]
+    (+ (.-offsetHeight element)
+       (js/parseInt (.-marginTop style))
+       (js/parseInt (.-marginBottom style)))))
+
+(defn set-visibility-ratio [frame element ratio]
+  (let [element-top (- (.-offsetTop element) (.-offsetTop frame))
+        element-height (true-height element) #_(.-offsetHeight element)
+        top-margin (/ (.-offsetHeight frame) 4)
+        position (+ element-top element-height (* -1 ratio element-height) (- top-margin))]
+    (.scrollTo frame 0 position)))
+
+(defn first-partially-visible-edit-field []
+  (let [fields (array-seq (.querySelectorAll js/document "#create-form .form-field"))
+        visible? #(<= 0 (-> % .getBoundingClientRect .-bottom))]
+    (first (filter visible? fields))))
+
+(defn autoscroll []
+  (let [edit-field (first-partially-visible-edit-field)
+        id (.getAttribute edit-field "data-field-id")
```

Comment:
first-partially-visible-edit-field can be null, which crashes here. For example after navigating to another page, or when creating a new form from scratch.